backend/services/event.py

```diff
@@ -0,0 +1,176 @@
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            event: a valid Event model representing the event to be added
+
+        Returns:
+            EventDetails: a valid EventDetails model representing the new Event
+        """
+        
+        # Ensure that the user has appropriate permissions to create users
+        self._permission.enforce(subject, "organization.events.manage", f"organization/{event.organization_id}")
+
+        # Checks if the role already exists in the table
+        if event.id:
+            event.id = None
+        
+        # Otherwise, create new object
+        event_entity = EventEntity.from_model(event)
+
+        # Add new object to table and commit changes
+        self._session.add(event_entity)
+        self._session.commit()
+
+        # Return added object
+        return event_entity.to_details_model()
+
+    def get_from_id(self, id: int) -> EventDetails:
+        """
+        Get the event from an id
+        If none retrieved, a debug description is displayed.
+
+        Parameters:
+            id: a valid int representing a unique event ID
+
+        Returns:
+            Event: Object with corresponding ID
+        """
+
+        # Query the event with matching id
+        entity = self._session.query(EventEntity).get(id)
+
+        # Check if result is null
+        if entity:
+            # Convert entry to a model and return
+            return entity.to_details_model()
+        else:
+            # Raise exception
+            raise EventNotFoundException(id);
+
+    def get_events_from_organization(self, id: int) -> list[EventDetails]:
+        """
+        Get all the events hosted by an organization with id
+
+        Parameters:
+            slug: a valid str representing a unique Organization slug
+
+        Returns:
+            list[EventDetail]: a list of valid EventDetails models
+        """
+
+        # Query the event with matching organization slug
+        events = self._session.query(EventEntity).filter(EventEntity.organization_id == id).all()
+        
+        # Convert entities to models and return
+        return [event.to_details_model() for event in events]
+
+    def update(self, subject: User, event: Event) -> EventDetails:
+        """
+        Update the event
+        If none found, a debug description is displayed.
+
+        Parameters:
+            event: a valid Event model
+
+        Returns:
+            EventDetails: a valid EventDetails model representing the updated event object
+        """
+        
+        # Ensure that the user has appropriate permissions to update users
+        self._permission.enforce(subject, "organization.events.manage", f"organization/{event.organization_id}")
+
+        # Query the event with matching id
+        obj = self._session.query(EventEntity).get(event.id)
+
+        # Check if result is null
+        if obj:

```

Comment:
I wonder if it'd be more idiomatic to check `if obj is not Null` and raise the exception, otherwise let the else branch be implicit from that point and not have an `else`.